test/unit/range/iterator_test_template.hpp

```diff
@@ -642,6 +651,88 @@ TYPED_TEST_P(iterator_fixture, compare_geq)
     }
 }
 
+// ---------------------------------------------------------------------------------------------------------------------
+// Contiguous Iterator
+// ---------------------------------------------------------------------------------------------------------------------
+
+// C++20 feature std::to_address
+// Implementation copied from: https://en.cppreference.com/w/cpp/memory/to_address
+template<class T>
+constexpr T* to_address(T* p) noexcept
+{
+    static_assert(!std::is_function_v<T>);
+    return p;
+}
+
+template<class T>
+constexpr auto to_address(const T& p) noexcept
+{
+    if constexpr (requires{ std::pointer_traits<T>::to_address(p); }) {
+        return std::pointer_traits<T>::to_address(p);
+    } else {
+        return to_address(p.operator->());
+    }
+}
```

Comment:
move to `seqan3/std/memory`

Reply:
Why is https://en.cppreference.com/w/cpp/memory/addressof not possible?

Reply:
> move to seqan3/std/memory

Yes! just testing if all compilers are fine with the implementation. This is rather a draft PR.
And actually it does not work :( I will investigate.

> Why is https://en.cppreference.com/w/cpp/memory/addressof not usable?

I just followed the standard proposal http://eel.is/c++draft/iterator.concept.contiguous